caproto/_data.py

```diff
@@ -219,60 +246,73 @@ def __init__(self, *, value=None, timestamp=None, status=0, severity=0,
                            )
         }
 
+    value = property(lambda self: self._data['value'])
+    timestamp = property(lambda self: self._data['timestamp'])
+
     def subscribe(self, queue, *sub_queue_args):
         '''Set subscription queue'''
         self._subscription_queue = queue
         self._subscription_queue_args = sub_queue_args
 
-    def fromtype(self, values, data_type):
-        '''Convenience function to convert given values to this data type'''
-        native_from = native_type(data_type)
-        return convert_values(values=values, from_dtype=native_from,
-                              to_dtype=self.data_type,
-                              string_encoding=self.string_encoding,
-                              enum_strings=getattr(self, 'enum_strings', None))
-
-    def astype(self, to_dtype):
-        '''Convenience function: convert stored data to a specific type'''
-        native_to = native_type(to_dtype)
-        return convert_values(values=self.value, from_dtype=self.data_type,
-                              to_dtype=native_to,
-                              string_encoding=self.string_encoding,
-                              enum_strings=getattr(self, 'enum_strings', None))
-
-    async def get_dbr_data(self, type_):
+    async def read(self, hostname, username, data_type):
```

Comment:
I like it but also feel that Python should be able to get a DBR type out without having to fake a host name and username